src/import-v1/index.ts

```diff
@@ -0,0 +1,377 @@
+        canonical: true,
+      };
+      this.push(subdomain);
+    }
+    callback();
+  }
+}
+
+async function readZones(zfname: string): Promise<Map<string, string>> {
+  const hashes = new Map<string, string>();
+
+  const zstream = stream.pipeline(fs.createReadStream(zfname), new LineReaderStream(), err => {
+    if (err) logError(`readzones: ${err}`);
+  });
+
+  const generator = asyncIterableToGenerator<string>(zstream);
+
+  while (true) {
+    const [keyRes, chunkRes] = [await generator.next(), await generator.next()];
+    if (!keyRes.done && !chunkRes.done) {
+      hashes.set(keyRes.value, chunkRes.value.replace(/\\n/g, '\n'));
+    } else {
+      break;
+    }
+  }
+
+  await finished(zstream);
+
+  return hashes;
+}
+
+async function valid(fileName: string): Promise<boolean> {
+  const shafname = `${fileName}.sha256`;
+  const hash = crypto.createHash('sha256');
+  await access(fileName, fs.constants.R_OK);
+  await access(shafname, fs.constants.R_OK);
+  const expected = (await readFile(shafname, { encoding: 'utf-8' })).trim();
+  await pipeline(fs.createReadStream(fileName), hash);
+  const calchash = hash.digest('hex');
+  if (expected !== calchash) {
+    logError(`calculated ${calchash} for ${fileName} != ${expected}`);
+    return false;
+  }
+  return true;
+}
+
+async function* readSubdomains(importDir: string) {
+  const metaIter = asyncIterableToGenerator<DbBNSSubdomain>(
+    stream.pipeline(
+      fs.createReadStream(path.join(importDir, 'subdomains.csv')),
+      new LineReaderStream({ highWaterMark: SUBDOMAIN_BATCH_SIZE }),
+      new SubdomainTransform(),
+      error => {
+        if (error) {
+          console.error('Error reading subdomains.csv');
+          console.error(error);
+        }
+      }
+    )
+  );
+
+  const zfIter = asyncIterableToGenerator<SubdomainZonefile>(
+    stream.pipeline(
+      fs.createReadStream(path.join(importDir, 'subdomain_zonefiles.txt')),
+      new LineReaderStream({ highWaterMark: SUBDOMAIN_BATCH_SIZE }),
+      new SubdomainZonefileParser(),
+      error => {
+        if (error) {
+          console.error('Error reading subdomain_zonefiles.txt');
+          console.error(error);
+        }
+      }
+    )
+  );
+
+  while (true) {
+    const meta = await metaIter.next();
+    const zf = await zfIter.next();
+    if (meta.done !== zf.done) {
+      throw new Error(
+        `Unexpected subdomain streams end mismatch; zonefiles ended: ${zf.done}, metadata ended: ${meta.done}`
+      );
+    }
+    if (meta.done || zf.done) {
+      break;
+    }
+    const subdomain = meta.value;
+    if (subdomain.zonefile_hash !== zf.value.hash) {
+      throw new Error(
+        `Unordered entries between subdomains.csv and subdomain_zonefiles.txt! Expected hash ${subdomain.zonefile_hash} got ${zf.value.hash}`
+      );
+    }
+    subdomain.zonefile = zf.value.content;
+    yield subdomain;
+  }
+}
+
+export async function importV1(db: PgDataStore, importDir?: string) {
```

Comment:
We need to add a unit test for this. In the `stacks-blockchain` repo this was done by using small samples of the import files, for example these:
* https://github.com/blockstack/stacks-blockchain/blob/master/stx-genesis/name_zonefiles-test.txt
* https://github.com/blockstack/stacks-blockchain/blob/master/stx-genesis/chainstate-test.txt

This allows the test to run quickly while still working with some real data. For this repo, we'd need those above 2 files, and a new `subdomain_zonefiles-test.txt` file. 

These few test files can be committed into a directory like `src/tests-bns/import-test-files/`, and the unit test would just call this `importV1` function with that directory, then a few simple BNS API lookups to ensure they imported into the db correctly.